modules/standard/online/online_controller.py

```diff
@@ -252,6 +304,21 @@ def get_online_alts_output(self, profession):
 
         return ChatBlob("%s Alts of Online Characters (%d)" % (profession, count), blob)
 
+    def get_logon(self, char_id):
+        content = self.db.query_single("SELECT logon FROM logmessages WHERE char_id=?", [char_id])
+        if content:
+            return "<grey>" + content.get("logon") + "<end>"
+        else:
+            return ""
+
+    def get_logoff(self, char_id):
+        content = self.db.query_single("SELECT logoff FROM logmessages WHERE char_id=?", [char_id])
+        if content:
+            return "<grey>" + content.get("logoff") + "<end>"
+        else:
+            return ""
+
+
```

Comment:
both get_logon() and get_logoff() should be in the LogController, and OnlineController should inject an instance of the LogController and call them from there